blocks/shopify/includes/options/store/class-store.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * Store
+ *
+ * @package Cata\Blocks\Shopify\Options
+ * @since 0.8.2
+ */
+
+namespace Cata\Blocks\Shopify\Options;
+
+use Cata\Blocks\Shopify\Options\Add_Store;
+
+/**
+ * Store
+ */
+class Store {
+
+	/**
+	 * Setting Name
+	 * Unique key for a single option/setting, stored in DB.
+	 */
+	const SETTING_NAME = 'cata_blocks_shopify_stores';
+
+	/**
+	 * Register
+	 * 
+	 * @param string $option_group
+	 */
+	public static function register( string $option_group ) : void {
+		register_setting(
+			$option_group,
+			self::SETTING_NAME,
+			array(
+				'type'              => 'string',
+				'sanitize_callback' => array( __CLASS__, 'sanitize_settings' ),
+				'show_in_rest'      => true,
+				'default'           => '',
+			)
+		);
+
+		add_settings_field(
+			self::SETTING_NAME,
+			'Shopify Stores',
+			array( __CLASS__, 'field_callback' ),
+			$option_group,
+			$option_group . '_section',
+			array(
+				'label_for' => self::SETTING_NAME,
+				'id'        => self::SETTING_NAME,
+				'name'      => self::SETTING_NAME,
+			)
+		);
+	}
+
+	/**
+	 * Sanitize Settings
+	 * 
+	 * @param array $array
+	 * @return array
+	 */
+	public static function sanitize_settings( $array ) : string {
+		if ( empty( $array ) ) {
+			return array();
+		}
+
+		$array = map_deep( $array, 'sanitize_text_field' );
+
+		$array = array_filter( $array, function( $value ) {
+			return ! empty( $value['subdomain'] ) && ! empty( $value['access_token'] );
+		} );
+
+		return wp_json_encode( $array );
+	}
+
+	/**
+	 * Field Callback
+	 * 
+	 * @param array $args Everything you need for an input.
+	 */
+	public static function field_callback( array $args ) : void {
+		$key = 0;
+		$setting = json_decode( get_option( self::SETTING_NAME, '' ), true );
```

Comment:
I get an error here when there's no data in this setting yet.

Doing `json_decode()` on an empty string returns null, which we can't `count()`

Using `'[]'` as the default value instead of `''` works since it decodes to an empty array.

It would also work to break up this one-liner so it can be type checked before trying to decode and count.

Reply:
I think `json_decode` only accepts a string, so `[]` being the default option value didn't work for me
https://www.php.net/manual/en/function.json-decode.php

But I can still check if the option is valid json before continuing

Reply:
Oh gotcha, I misunderstood!